src/cascade/core/db.py

```diff
@@ -0,0 +1,55 @@
+from contextlib import contextmanager
+
+try:
+    from db_tools import ezfuncs
+except ImportError:
+
+    class DummyEZFuncs:
+        def __getattr__(self, name):
+            raise ImportError(f"Required package db_tools not found")
+
+    ezfuncs = DummyEZFuncs()
+
+
+@contextmanager
+def cursor(execution_context=None, database=None):
+    """A context manager which exposes a database cursor connected to the database specified by
+    the execution_context. The cursor will be closed when the manager exits and if it exits without
+    raising an exception the connection will also be commited.
+    """
+
+    if execution_context is None:
+        if database is None:
+            raise ValueError("Must supply either execution_context or database")
+    else:
+        if database is not None:
+            raise ValueError("Must not supply both execution_context and database")
+        database = execution_context.parameters.database
+
```

Comment:
If context_manager wasn't supplied, and database was, database will be used here.

Reply:
Yep. This is because in the majority of cases we want the dismod_at database but occasionally we will need to access other databases, like the central epi database. I've had some discussions with CC since writing this which make it seem like we may eventually stop needing to access the epi database directly but for now it's still required.